api/v1/loadtest_types.go

```diff
@@ -161,6 +161,7 @@ type Driver struct {
 
 	// Run describes the run container, which is the runtime of the driver for
 	// the actual test.
+	// +optional
 	Run Run `json:"run"`
```

Comment:
Hold up a moment. This is what I meant in the meeting last week. Run cannot be optional, b/c you must always have a container to run.... at least an image. We always need a container to run.

In addition, this change violates the Kubernetes API conventions... we cannot make a non-pointer field `//+optional`. And optionals should have a `json:"run,omitempty"` tag. So this becomes `*Run` which is the fix we don't want and will break things later down the pipeline.

We need to discuss alternative ways to address this issue.

Reply:
> We need to discuss alternative ways to address this issue.

The run field is also required in the client and server, but that doesn't require us to include an image. The alternative solution is to make the driver behave in the same way as the clients and servers, and I agree that it does not require making the run field optional.

Reply:
It looks like "image" is optional. So the question may be what is the difference between the handling of image in these two places:

https://github.com/grpc/test-infra/blob/de9cdd98b55cb95db97b13356d2cacf10dbb1bf2/config/defaults.go#L195

https://github.com/grpc/test-infra/blob/de9cdd98b55cb95db97b13356d2cacf10dbb1bf2/config/defaults.go#L216

Reply:
One issue with the above snippets is that they are picking driver and client images from the same image map, and that cannot be right.

Reply:
Here is something interesting: if I make an assumption about the 'command' to run the driver, I can specify the name without specifying the image, and the test succeeds:

```yaml
  driver:
    language: cxx
    name: '0'
    run:
      command:
      - /src/driver/run.sh
```

Reply:
It looks like it is possible to make it work with no assumptions, by always adding `driver`, `run` and `env` fields if they don't exist, resulting in the following: 

```yaml
  driver:
    language: cxx
    name: '0'
    run:
      env: []
```
I will update the config generator and the examples.

We still need to fix the logic that might apply worker images to the driver.

Reply:
FYI, the final solution is in #189:

```yaml
  driver:
    language: cxx
    name: '0'
    run: {}
```

Reply:
Thank you for the discussion. Currently in Driver struct, Run is a required filed, but every field inside Run struct itself is optional. So we could get aways with adding `run: {}` as mentioned by @paulosjca . @codeblooded Thank you for helping me learning the [Kubernetes's conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).
I closed this PR since it is better addressed in #189.